pennylane/transforms/transpile.py

```diff
@@ -0,0 +1,212 @@
+        measurements = expanded_tape.measurements.copy()
+
+        while len(list_op_copy) > 0:
+            op = list_op_copy[0]
+
+            # gates which act only on one wire
+            if op.num_wires != 2:
+                gates.append(op)
+                list_op_copy.pop(0)
+                continue
+
+            # two-qubit gates which can be handled by the coupling map
+            if (
+                op.wires in coupling_graph.edges
+                or tuple(reversed(op.wires)) in coupling_graph.edges
+            ):
+                gates.append(op)
+                list_op_copy.pop(0)
+                continue
+
+            # since in each iteration, we adjust indices of each op, we reset logical -> phyiscal mapping
+            map_wires = {w: w for w in tape.wires}
+
+            # to make sure two qubit gates which act on non-neighbouring qubits q1, q2 can be applied, we first look
+            # for the shortest path between the two qubits in the connectivity graph. We then move the q2 into the
+            # neighbourhood of q1 via swap operations.
+            source_wire, dest_wire = op.wires
+            shortest_path = nx.algorithms.shortest_path(coupling_graph, source_wire, dest_wire)
+            path_length = len(shortest_path) - 1
+            wires_to_swap = [shortest_path[(i - 1) : (i + 1)] for i in range(path_length, 1, -1)]
+
+            for w0, w1 in wires_to_swap:
+                # swap wires
+                gates.append(SWAP(wires=[w0, w1]))
+                # update logical -> phyiscal mapping
+                map_wires = {
+                    k: (w0 if v == w1 else (w1 if v == w0 else v)) for k, v in map_wires.items()
+                }
+
+            # append op to gates with adjusted indices and remove from list
+            gates.append(_adjust_op_indices(op, map_wires))
+            list_op_copy.pop(0)
+
+            # adjust qubit indices in remaining ops + measurements to new mapping
+            list_op_copy = [_adjust_op_indices(op, map_wires) for op in list_op_copy]
+            measurements = [_adjust_mmt_indices(m, map_wires) for m in measurements]
+
+    for op in gates + measurements:
+        apply(op)
+
+
+def _adjust_op_indices(_op, _map_wires):
+    """helper function which adjusts wires in Operation according to the map _map_wires"""
+    _new_wires = Wires([_map_wires[w] for w in _op.wires])
+    _params = _op.parameters
+    if len(_params) == 0:
+        return type(_op)(wires=_new_wires)
+    return type(_op)(*_params, wires=_new_wires)
+
+
+def _adjust_mmt_indices(_m, _map_wires):
+    """helper function which adjusts wires in MeasurementProcess according to the map _map_wires"""
+    _new_wires = Wires([_map_wires[w] for w in _m.wires])
+
+    if isinstance(_m.obs, (Hamiltonian, Tensor)):
+        # tensor products of observables
+        raise NotImplementedError(
+            "Measuring expectation values of tensor products or Hamiltonians is not yet supported"
+        )
```

Comment:
@mauriceweber I don't think this error can ever be raised, since such cases should be caught by the exception on line 94? In which case, this can be removed :)